src/sql/parts/modelComponents/queryTextEditor.ts

```diff
@@ -119,7 +119,20 @@ export class QueryTextEditor extends BaseTextEditor {
 		if (!this._config) {
 			this._config = new Configuration(undefined, editorWidget.getDomNode());
 		}
-		let editorHeightUsingLines = this._config.editor.lineHeight * editorWidget.getModel().getLineCount();
+		let editorWidgetModel = editorWidget.getModel();
+		let lineCount = editorWidgetModel.getLineCount();
+		// Need to also keep track of lines that wrap; if we just keep into account line count, then the editor's height would not be
+		// tall enough and we would need to show a scrollbar. Unfortunately, it looks like there isn't any metadata saved in a ICodeEditor
+		// around max column length for an editor (which we could leverage to see if we need to loop through every line to determine
+		// number of lines that wrap). Finally, viewportColumn is calculated on editor resizing automatically; we can use it to ensure
+		// that the viewportColumn will always be greater than any character's column in an editor.
+		let numberWrappedLines = 0;
+		for(let line = 1; line <= lineCount; line++) {
+			if (editorWidgetModel.getLineMaxColumn(line) >= this._config.editor.layoutInfo.viewportColumn - 1) {
```

Comment:
Wow. This is cool, are we sure the logic will always be "correct"? Any edge cases we can think of / anywhere else using similar logic?

Reply:
I tried:
- copying/pasting a bunch of code inline
- opening files with tons of characters per line
- opening files with just barely over the number of characters before it wrapped
- lines with tons of spaces

Couldn't get it to break (and show the scrollbar), but doesn't mean there aren't bugs :)